context.go

```diff
@@ -248,3 +248,19 @@ func (c *Ctx) AbsForImport(path string) (string, error) {
 	}
 	return posspath, nil
 }
+
+// ValidateParams ensure that solving can be completed with the specified params.
+func (c *Ctx) ValidateParams(sm gps.SourceManager, params gps.SolveParameters) error {
+	err := gps.ValidateParams(params, sm)
+	if err != nil {
+		if deduceErrs, ok := err.(gps.DeductionErrs); ok {
+			c.Err.Println("The following errors occurred while deducing packages:")
+			for ip, dErr := range deduceErrs {
+				c.Err.Printf("  * \"%s\": %s", ip, dErr)
+			}
+			c.Err.Println()
+		}
+	}
+
+	return errors.Wrap(err, "validateParams")
+}
```

Comment:
Since all the parts involved in this are gps related, maybe adding it to [gps/solver.go](https://github.com/golang/dep/blob/master/internal/gps/solver.go) would be better, like `ValidateParamsWithLogs()` or some better name? Just a suggestion.